ams/server/src/main/java/com/netease/arctic/server/table/TableRuntime.java

```diff
@@ -238,10 +245,36 @@ public void completeProcess(boolean success) {
   }
 
   private void updateOptimizingStatus(OptimizingStatus status) {
+    SelfOptimizingReport selfOptimizingReport = new SelfOptimizingReport(tableIdentifier.toString());
+    switch (optimizingStatus) {
+      case IDLE:
+        selfOptimizingReport.tableOptimizingIdleDurationSecond()
+            .inc(currentStatesDurationSecond());
+        metricsReporters.report(selfOptimizingReport);
+        break;
+      case PENDING:
+        selfOptimizingReport.tableOptimizingPendingDurationSecond()
+            .inc(currentStatesDurationSecond());
+        metricsReporters.report(selfOptimizingReport);
+        break;
+      case COMMITTING:
+        selfOptimizingReport.setOptimizingType(optimizingProcess.getOptimizingType().name());
+        selfOptimizingReport.setOptimizingProcessId(optimizingProcess.getProcessId());
+        selfOptimizingReport.setTargetSnapshotId(optimizingProcess.getTargetSnapshotId());
+        selfOptimizingReport.setOptimizingStatus(optimizingProcess.getStatus().name());
+        selfOptimizingReport.tableOptimizingCommitDurationSecond()
+            .inc(currentStatesDurationSecond());
```

Comment:
We can use Timer.time(() -> {//do something}) to calculate the cost time of a certain operation.

Reply:
1